sgkit/utils.py

```diff
@@ -336,3 +336,22 @@ def hash_array(x: ArrayLike, out: ArrayLike) -> None:  # pragma: no cover
     out[0] = 5381
     for i in range(x.shape[0]):
         out[0] = out[0] * 33 + x[i]
+
+
+def smallest_numpy_int_dtype(value: int) -> Optional[DType]:
+    """Return the smallest NumPy signed integer dtype that can be used to store the given value.
+
+    Parameters
+    ----------
+    value
+        An integer value to be stored.
+
+    Returns
+    -------
+    A NumPy signed integer dtype suitable for storing the given value, or None if the value exceeds
+    the bounds of the largest dtype (``np.int64``).
+    """
+    for dtype in (np.int8, np.int16, np.int32, np.int64):
+        if np.iinfo(dtype).min <= value <= np.iinfo(dtype).max:
+            return dtype
+    return None
```

Comment:
This return None case basically can't happen, so why not raise an OverflowError here rather than requiring client code check the return value (C style)?

Reply:
Yes, that's much better. I've implemented this in https://github.com/pystatgen/sgkit/pull/667/commits/f63600f2a15d40663af05d0b934061229c50b3e1. Should be ready to go in now.